docs/route/route.md

```diff
@@ -327,6 +328,56 @@ Please see the [errorElement][errorelement] documentation for more details.
 
 Any application-specific data. Please see the [useMatches][usematches] documentation for details and examples.
 
+## `lazy`
+
+In order to keep your application bundles small and support code-splitting of your routes, each route can provide an async function that resolves the non-route-matching portions of your route definition (`loader`, `action`, `element`, `errorElement`, etc.).
```

Comment:
Should we trim this down to a short example and link off to the above doc for the in-depth explanation?

Reply:
I've trimmed it a bit, let me know what you think.